cmd/osv-scanner/main_test.go

```diff
@@ -509,6 +509,209 @@ func TestRun_LockfileWithExplicitParseAs(t *testing.T) {
+				Scanned %%/fixtures/locks-one-with-nested/nested/composer.lock file and found 1 packages
+				Scanned %%/fixtures/locks-one-with-nested/yarn.lock file and found 1 packages
+				No vulnerabilities found
+			`,
+			wantStderr: "",
+		},
+		// .gitignored files
+		{
+			name:         "",
+			args:         []string{"", "-experimental-local-db", "--recursive", "./fixtures/locks-gitignore"},
+			wantExitCode: 0,
+			wantStdout: `
+				Scanning dir ./fixtures/locks-gitignore
+				Scanned %%/fixtures/locks-gitignore/Gemfile.lock file and found 1 packages
+				Scanned %%/fixtures/locks-gitignore/subdir/yarn.lock file and found 1 packages
+				No vulnerabilities found
+			`,
+			wantStderr: "",
+		},
+		// ignoring .gitignore
+		{
+			name:         "",
+			args:         []string{"", "-experimental-local-db", "--recursive", "--no-ignore", "./fixtures/locks-gitignore"},
+			wantExitCode: 0,
+			wantStdout: `
+				Scanning dir ./fixtures/locks-gitignore
+				Scanned %%/fixtures/locks-gitignore/Gemfile.lock file and found 1 packages
+				Scanned %%/fixtures/locks-gitignore/composer.lock file and found 1 packages
+				Scanned %%/fixtures/locks-gitignore/ignored/Gemfile.lock file and found 1 packages
+				Scanned %%/fixtures/locks-gitignore/ignored/yarn.lock file and found 1 packages
+				Scanned %%/fixtures/locks-gitignore/subdir/Gemfile.lock file and found 1 packages
+				Scanned %%/fixtures/locks-gitignore/subdir/composer.lock file and found 1 packages
+				Scanned %%/fixtures/locks-gitignore/subdir/yarn.lock file and found 1 packages
+				Scanned %%/fixtures/locks-gitignore/yarn.lock file and found 1 packages
+				No vulnerabilities found
+			`,
+			wantStderr: "",
+		},
+		// output with json
+		{
+			name:         "",
+			args:         []string{"", "-experimental-local-db", "--json", "./fixtures/locks-many/composer.lock"},
+			wantExitCode: 0,
+			wantStdout: `
+				{
+					"results": []
+				}
+			`,
+			wantStderr: `
+				Scanning dir ./fixtures/locks-many/composer.lock
+				Scanned %%/fixtures/locks-many/composer.lock file and found 1 packages
+			`,
+		},
+		{
+			name:         "",
+			args:         []string{"", "-experimental-local-db", "--format", "json", "./fixtures/locks-many/composer.lock"},
+			wantExitCode: 0,
+			wantStdout: `
+				{
+					"results": []
+				}
+			`,
+			wantStderr: `
+				Scanning dir ./fixtures/locks-many/composer.lock
+				Scanned %%/fixtures/locks-many/composer.lock file and found 1 packages
+			`,
+		},
+		// output format: markdown table
+		{
+			name:         "",
+			args:         []string{"", "-experimental-local-db", "--format", "markdown", "./fixtures/locks-many/composer.lock"},
+			wantExitCode: 0,
+			wantStdout: `
+				Scanning dir ./fixtures/locks-many/composer.lock
+				Scanned %%/fixtures/locks-many/composer.lock file and found 1 packages
+				No vulnerabilities found
+			`,
+			wantStderr: "",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
Running these in parallel causes the tests to fail on the initial run. I think it's because it tries to open the zip while it's being written to/updating.

We probably need some sort of locking mechanism to indicate that another process is currently updating the database. This can probably come in a followup PR though, and for now just removing the `t.Parallel()` should be good enough.

Reply:
I'm not sure if that is what's happening, because it shouldn't be creating the zip file in the first place if its 404ing.

However overall this does at least give a reason to support a flag for configuring the db location in addition to the env variable because then we'd be able to use a temp directory for each test (since you can't set env variables when running in parallel)

Reply:
Yup, this was [a bug](https://github.com/google/osv-scanner/pull/183/commits/617e10d23ca653e9c8b11cf97c705bc007e8588a)